### PR TITLE
Issue #986 Isolated stackable business logic

### PR DESCRIPTION
**Pull request by mmoreram**
* Removed logic from event listener and added to a service
* Tested class with unit test
* Tested event listener with functional tests

### CODE REVIEW

File: src/Elcodi/Component/CartCoupon/Tests/Services/StackableCouponCheckerTest.php [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+     */
+    public function dataCheckStackableCoupon()
+    {
+        return [
+            [true, [], false],
+            [false, [], false],
+            [false, [true, true], true],
+            [false, [false, true], true],
+            [true, [false, true], true],
+            [true, [true, true], false],
```
Comment (EmanueleMinotto) [inline src/Elcodi/Component/CartCoupon/Tests/Services/StackableCouponCheckerTest.php]: Imo adding a comment here for every case could be useful for the future. :)

Comment (mmoreram) [inline src/Elcodi/Component/CartCoupon/Tests/Services/StackableCouponCheckerTest.php]: Updated to both tests data providers :)

**Comment by EmanueleMinotto** [comment]
this is great, congrats!

**Comment by mmoreram** [comment]
@EmanueleMinotto Thanks :dancer:

**Pull request merged**